src/cryptonote_core/service_node_list.cpp

```diff
@@ -915,9 +916,9 @@ namespace service_nodes
                               });
       if (cit != contributor.locked_contributions.end())
       {
-        if (hf_version >= hf::hf19_reward_batching)
+        if (hf_version >= hf::hf20)
         {
-          if (cit->amount < service_nodes::SMALL_CONTRIBUTOR_THRESHOLD && (block_height - node_info.registration_height) < service_nodes::SMALL_CONTRIBUTOR_UNLOCK_TIMER)
+          if (cit->amount < small_contributor_amount_threshold && (block_height - node_info.registration_height) < service_nodes::SMALL_CONTRIBUTOR_UNLOCK_TIMER)
```

Comment:
The original condition/check needs to stay for hf19 because it would open up sync issues before old/new.  E.g. if I want to cause havoc, I got register a node with a miniscule reserved spot (<3749), submit it, stake it, then request an unlock: upgraded nodes would accept it, but non-upgraded nodes would reject it here.

Reply:
Out of curiosity, why is the small contributor check `< 3749`, rather than `< 3750`?

Reply:
We were getting some integer rounding errors cause contributions internally were represented as an integer over `inline constexpr uint64_t STAKING_PORTIONS = UINT64_C(0xfffffffffffffffc);` (We have since changed this, partially because these rounding errors were becoming annoying).

The extra oxen is just a conservative buffer cause contributing 3750 oxen would result in a having a contribution percentage of 24.99999999%

Although that wasn't an issue here when we were directly comparing the amount with 3749. But that had its own issues as you can see

Reply:
Thank you for the detailed explanation, Sean.